packages/@aws-cdk/aws-pipes-sources-alpha/lib/kinesis.ts

```diff
@@ -0,0 +1,161 @@
+   * @default no batching window
+   */
+  readonly maximumBatchingWindow?: Duration;
+
+  /**
+   * (Streams only) Discard records older than the specified age. The default value is -1, which sets the maximum age to infinite. When the value is set to infinite, EventBridge never discards old records.
+   *
+   * Leave undefined to set the maximum record age to infinite.
+   *
+   *  @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-maximumrecordageinseconds
+   * @default -1 (infinite)
+   */
+  readonly maximumRecordAge?: Duration;
+
+  /**
+   * (Streams only) Discard records after the specified number of retries. The default value is -1, which sets the maximum number of retries to infinite. When MaximumRetryAttempts is infinite, EventBridge retries failed records until the record expires in the event source.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-maximumretryattempts
+   * @default -1 (infinite)
+   */
+  readonly maximumRetryAttempts?: number;
+
+  /**
+   * (Streams only) Define how to handle item process failures. AUTOMATIC_BISECT halves each batch and retry each half until all the records are processed or there is one failed message left in the batch.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-onpartialbatchitemfailure
+   * @default off
+   */
+  readonly onPartialBatchItemFailure?: OnPartialBatchItemFailure;
+
+  /**
+   * (Streams only) The number of batches to process concurrently from each shard. The default value is 1.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-parallelizationfactor
+   * @default 1
+   */
+  readonly parallelizationFactor?: number;
+
+  /**
+   * (Streams only) The position in a stream from which to start reading.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-startingposition
+   */
+  readonly startingPosition: KinesisStartingPosition;
+
+  /**
+   * With StartingPosition set to AT_TIMESTAMP, the time from which to start reading, in Unix time seconds.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-startingpositiontimestamp
+   * @default no starting position timestamp
+   */
+  readonly startingPositionTimestamp?: string;
```

Comment:
Pretty sure this can be a number. I'd also add `{@link}`s to refer to the other field/value

Reply:
CDK seems to go back and forth on this, depending what CFN expects..

Reply:
Up to you, but I think you can tell that a `number` timestamp field is an epoch without looking at the JSDoc. A `string` timestamp, I wouldn't be sure if it's an epoch or an ISO 8601 date/datetime at first glance